okhttp/src/main/java/io/grpc/okhttp/OkHttpClientTransport.java

```diff
@@ -439,6 +464,69 @@ public void close() {}
     return null;
   }
 
+  private Socket createHttpProxySocket(InetSocketAddress address, InetSocketAddress proxyAddress,
+      String proxyUsername, String proxyPassword) {
+    Status internalStatus = Status.INTERNAL.withDescription("Failed trying to connect with proxy");
+    try {
+      Socket sock = new Socket(proxyAddress.getAddress(), proxyAddress.getPort());
+      sock.setTcpNoDelay(true);
+
+      BufferedSource source = Okio.buffer(Okio.source(sock));
+      BufferedSink sink = Okio.buffer(Okio.sink(sock));
+
+      // Prepare headers and request method line
+      Request proxyRequest = createHttpProxyRequest(address, proxyUsername, proxyPassword);
+      HttpUrl url = proxyRequest.httpUrl();
+      String requestLine = String.format("CONNECT %s:%d HTTP/1.1", url.host(), url.port());
+
+      // Write request to socket
+      sink.writeUtf8(requestLine).writeUtf8("\r\n");
+      for (int i = 0, size = proxyRequest.headers().size(); i < size; i++) {
+        sink.writeUtf8(proxyRequest.headers().name(i))
+            .writeUtf8(": ")
+            .writeUtf8(proxyRequest.headers().value(i))
+            .writeUtf8("\r\n");
+      }
+      sink.writeUtf8("\r\n");
+      // Flush buffer (flushes socket and sends request)
+      sink.flush();
+
+      // Read response, parse headers, check if 200 OK was returned
+      StatusLine statusLine = StatusLine.parse(source.readUtf8LineStrict());
+      if (statusLine.code != 200) {
```

Comment:
RFC 2817§5.3 says any 2xx response is successful. I do see that Netty is apparently accepting 200 only. It looks like [C core is following the RFC](https://github.com/grpc/grpc/blob/master/src/core/ext/client_channel/http_connect_handshaker.c#L168).